web/src/components/navbar/NavBarPopupMenu.tsx

```diff
@@ -0,0 +1,188 @@
+import { ButtonIcon } from "@/components/ButtonIcon";
+import { NavMenuItem } from "@/components/navbar/NavMenuItem";
+import { Icon } from "@/components/njwds/Icon";
+import { AuthAlertContext } from "@/contexts/authAlertContext";
+import { AuthContext } from "@/contexts/authContext";
+import { triggerSignIn } from "@/lib/auth/sessionHelper";
+import { onSelfRegister, onSignOut } from "@/lib/auth/signinHelper";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import analytics from "@/lib/utils/analytics";
+import { getNavBarBusinessTitle, getUserNameOrEmail } from "@/lib/utils/helpers";
+import Config from "@businessnjgovnavigator/content/fieldConfig/config.json";
+import { MenuItem, MenuList } from "@mui/material";
+import { useRouter } from "next/router";
+import { ReactElement, useContext } from "react";
+
+export type menuConfiguration =
+  | "profile"
+  | "profile-register-login"
+  | "profile-myNJ-logout"
+  | "login"
+  | "login-getstarted";
```

Comment:
A few thoughts:
- did you consider naming these after states instead of render results? Looking at them, it's hard to decipher when I might want to use "profile" vs "profile-myNJ-logout" etc. I think it might be a lot easier for someone else to use these if they described when they get used. "onboarding" and "guest-mode" and so on. Otherwise, I have to find the actual logical if/else structure just to understand the difference between when they show up.
- in any case, it'd be nice to standardize the casing/style of these terms. It's almost kebab-case between words except for "getstarted". And it's almost all lowercase except for "myNJ"

Reply:
So I had considered it the issue being largely that it isn't consistent across mobile and desktop in some but not all cases which would mean our division would be something like.

landing - mobile (landing desktop doesn't exist)
onboarding - mobile (onboarding desktop doesn't exist)
guest - mobile
guest - desktop 
authenticated - mobile & desktop

I also think this is more helpful to read in the NavBarMobile and NavBarDesktop components because you know when you render the NavBarPopupMenu exactly which parts you're getting based on the logic.